File: Api/src/test/java/net/binggl/mydms/test/tags/TagModelTest.java
package net.binggl.mydms.test.tags;

import static io.dropwizard.testing.FixtureHelpers.*;
import static org.assertj.core.api.Assertions.assertThat;
import org.junit.Test;

import com.fasterxml.jackson.databind.ObjectMapper;

import io.dropwizard.jackson.Jackson;
import net.binggl.mydms.tags.Tag;

public class TagModelTest {

	private static final ObjectMapper MAPPER = Jackson.newObjectMapper();

    @Test
    public void serializesToJSON() throws Exception {
        final Tag tag = new Tag(1, "tag1");

        final String expected = MAPPER.writeValueAsString(
                MAPPER.readValue(fixture("fixtures/tag.json"), Tag.class));

        assertThat(MAPPER.writeValueAsString(tag)).isEqualTo(expected);
    }
    
    @Test
    public void deserializesFromJSON() throws Exception {
        final Tag tag = new Tag(1, "tag1");
        assertThat(MAPPER.readValue(fixture("fixtures/tag.json"), Tag.class))
                .isEqualTo(tag);
    }
}

File: Api/src/test/kotlin/net/binggl/mydms/features/tags/TagModelTest.kt
package net.binggl.mydms.features.tags

import FixtureHelpers
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import org.junit.Assert
import org.junit.Test

class TagModelTest {

    @Test
    fun serializeToJson(): Unit {
        val tag = Tag(id = 1, name = "tag1")
        val expected = MAPPER.writeValueAsString(MAPPER.readValue(FixtureHelpers.fixture("fixtures/tag.json"), Tag::class.java))
        Assert.assertEquals(expected, MAPPER.writeValueAsString(tag))
    }

    @Test
    fun deserializeFromJson(): Unit {
        val tag = Tag(id = 1, name = "tag1")
        Assert.assertEquals(MAPPER.readValue(FixtureHelpers.fixture("fixtures/tag.json"), Tag::class.java), tag)
    }

    companion object {
        // https://github.com/FasterXML/jackson-module-kotlin
        private val MAPPER = jacksonObjectMapper()
    }
}